core/services/headtracker/head_broadcaster.go

```diff
@@ -68,23 +65,45 @@ func (hr *HeadBroadcaster) Start() error {
 
 func (hr *HeadBroadcaster) Close() error {
 	return hr.StopOnce("HeadBroadcaster", func() error {
+
+		for _, unsubscribe := range hr.toUnsubscribe {
+			unsubscribe()
+		}
+
 		close(hr.chClose)
 		hr.wgDone.Wait()
 		return nil
 	})
 }
 
 func (hr *HeadBroadcaster) Connect(head *models.Head) error {
+	hr.mutex.RLock()
+	callbacks := hr.callbacks.clone()
+	hr.mutex.RUnlock()
+
+	for i, callback := range callbacks {
+		err := callback.Connect(head)
+		if err != nil {
+			logger.Errorf("HeadBroadcaster: Failed Connect callback at index %v: %v", i, err)
+		}
+	}
+
 	return nil
 }
 
-func (hr *HeadBroadcaster) Disconnect() {}
-
 func (hr *HeadBroadcaster) OnNewLongestChain(ctx context.Context, head models.Head) {
 	hr.mailbox.Deliver(head)
 }
 
-func (hr *HeadBroadcaster) Subscribe(callback HeadBroadcastable) (unsubscribe func()) {
+func (hr *HeadBroadcaster) SubscribeUntilClose(callback httypes.HeadTrackable) {
```

Comment:
I wonder if you could merge SubscribeUntilClose and Subscribe? Unsubscribe when broadcaster closes, but also provide an optional `unsubscribe` callback that could be used to unsubscribe early.